allennlp/modules/vision/grid_embedder.py

```diff
@@ -1,127 +1,69 @@
-from typing import Union
+from collections import OrderedDict
+from typing import Tuple
 
-import torch
 from torch import nn, FloatTensor, IntTensor
+import torchvision
 
 from allennlp.common.registrable import Registrable
 
 
 class GridEmbedder(nn.Module, Registrable):
     """
-    A `GridEmbedder` takes a batch of images as a tensor with the dimensions
-    (Batch, Color, Height, Width), and returns a tensor in the format
-    (Batch, Features, new_height, new_width).
-
-    For every image, it embeds a patch of the image, and returns the embedding
-    of the patch. The size of the image might change during this operation.
+    A `GridEmbedder` takes a batch of images as a tensor with shape
+    `(batch_size, color_channels, height, width)`, and returns an ordered dictionary
+    of tensors with shape `(batch_size, *)`, each representing a specific feature.
     """
 
-    def forward(self, images: FloatTensor, sizes: IntTensor) -> FloatTensor:
+    def forward(self, images: FloatTensor, sizes: IntTensor) -> "OrderedDict[str, FloatTensor]":
```

Comment:
`OrderedDict`?

Reply:
See https://mypy.readthedocs.io/en/latest/common_issues.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime.

Reply:
I'm not confused about the string form of the type. I'm confused why it has to be an `OrderedDict` instead of a `Dict`.

Reply:
This is consistent with `torchvision`. I'm guessing it's because there is a natural order to the features, as each feature comes from a different layer of the model.